server/api/controller/cron.js

```diff
@@ -145,6 +146,86 @@ const sendPlanningAlerts = () => {
 		});
 };
 
+const planToEmail = (plan, map) => {
+	return {
+		id: plan.get('id'),
+		title: plan.get('plan_display_name'),
+		city: plan.get('PLAN_COUNTY_NAME'),
+		text: plan.get('goals_from_mavat'),
+		status: plan.get('status'),
+		areaChange: plan.describeHousingChange(),
+		map: 'data:image/gif;base64,'+ map,
+		link:`${this.baseUrl}plan/${plan.get('id')}`
+	};
+}; 
+
+const alertToEmail = (alert) => {
+	const alertTitle = `תוכניות חדשות בקרבת ${alert.get('address') ||  'תחומי הענין שלך'}`;
+	return {
+		alert:{
+			title: alertTitle,
+			unsubscribeLink: `${this.baseUrl}alerts/unsubscribe/${alert.unsubsribeToken()}`
+		}
+	};
+};
+
+const sendDigestPlanningAlerts = async () => {
```

Comment:
so to my understanding the logic is as follows (correct me if i'm wrong) -
1. query db for an alert that wasn't checked in X days
2. query db for Y plans that were created since the alert was last checked
3. for each plan generate static map and format other data
4. send alert-specific email to the user the alert belongs to
5. update alert last-checked field to time when this process began

the main technical issue i can think of here is that this process involves creating a static map per every plan (that will be sent) per every alert. this may strain the tile service (osm) and also our servers (rendering tiles together plus plan polygons) and might cause some issues, plus it will take a long time to run.
maybe a map cache can be used?
this will be somewhat mitigated by the fact that not every plan is sent (since only Y plans are queried).

product questions:
1. is it okay that a plan might be sent more than once to a user (if it matches more than one alert)?
2. also, is it okay that some plans will never be sent to a user (and will the plan selection be based on any plan parameters, or will it just be the Y newest plans)?

Reply:
You got the logic right. 
As per your first question, that is possible. Yet, we're working on a smarter digest, that will include all users alerts, so plan will be sent twice.

As per the second point, to prevent that, I'll update the last_updated of the email to be the latest date of the plans, and will fetch the plans by order. That means, that if your alert has exceeded 5 plans, you'll get the top 5 , and then, in another email, the rest. 
That is because sendgrid doesn't support arrays and loops. So I leave placeholder for 5 plans, but can increase it manually up to 10 if needed